py/visdom/server.py

```diff
@@ -840,6 +845,29 @@ def post(self):
         )
         self.wrap_func(self, args)
 
+class ForkHandler(BaseHandler):
+    def initialize(self, app):
+        self.app = app
+        self.state = app.state
+        self.login_enabled = app.login_enabled
+
+    @staticmethod
+    def wrap_func(handler, args):
+        prev_eid = args.get('prev_eid')
+        eid = args.get('eid')
+
+        if prev_eid not in handler.state:
+            return handler.write('TODO: env doesn\'t exit')
```

Comment:
You can use an assert in this case, the error should propagate to the visdom client, which I think is an appropriate outcome to trying to fork an environment that doesn't exist. Note in the docs that such an exception will occur if an env that doesn't exist is forked.